### PR TITLE
Handle substring in string itself in IDE0057

### LINKED ISSUE #38055: IDE0057 incorrectly handles substrings on String itself
**Version Used**: 
3.3.0-beta3-19413-06+ac06df1bffb7b7fd0e5bf63cc91921af76b21e03

**Steps to Reproduce**:
This is a super corner-case only relevant to a few people in the world, so I'll understand if you want to close it.  Just opening the issue in case.

The definition of System.String itself uses its own Substring in a few places, e.g.
```C#
splitStrings[arrIndex++] = Substring(currIndex, sepList[i] - currIndex);
```
IDE0057 fires here, and transforms it into:
```C#
splitStrings[arrIndex++] = Substring[currIndex..sepList[i]];
```
which doesn't compile.  It should have been transformed instead into:
```C#
splitStrings[arrIndex++] = this[currIndex..sepList[i]];
```
which works fine.

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Fixes #38055

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs [hunk 1]
```diff
@@ -119,6 +119,12 @@ private static ExpressionSyntax FixOne(Result result, SyntaxGenerator generator)
                     return ElementBindingExpression(argumentList);
                 }
 
+                if (invocation.Expression is IdentifierNameSyntax)
+                {
+                    // Substring(...) -> this[...]
+                    return ElementAccessExpression(ThisExpression(), argumentList);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs:125]: i'd prefer we also validate that we're inside the string class.  Or, if htis is handling more than just being inside System.String and is handling things like .Slice inside other types, then we can have a simpler test that just shows that.

Comment (huoyaoyuan) [reply src/Analyzers/CSharp/CodeFixes/UseIndexOrRangeOperator/CSharpUseRangeOperatorCodeFixProvider.cs:125]: I'd prefer the latter, adding a test for slicing on self.

**Comment by huoyaoyuan** [comment]
This one should be simple enough. Not sure what the CI failure is.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**